code_files/report_calculator.py

```diff
@@ -0,0 +1,49 @@
+from code_files.weather_readings import WeatherExtremes
+
+
+class ReportCalculator:
+    def __init__(self, readings):
+        self.readings = readings
+    
+    def compute_extreme_stats(self):
+        """Computes the extreme weather statistics for a particular year"""
+        
+        highest_temp_reading = max(
+            self.readings,
+            key=lambda r: r.max_temp if r.max_temp is not None else float("-inf"),
```

Comment:
Do we need this check now that we are not parsing empty rows?

Reply:
This check is required. There are certain rows in which some values are missing, and they are read as `None`. If this check is removed, we get the following error:
```
Traceback (most recent call last):
  File "/Users/tayyab.tahir/Documents/MTTQ/Training/Task1_WeatherMan/weatherman.py", line 27, in <module>
    weather_extremes = report_calculator.compute_extreme_stats()
  File "/Users/tayyab.tahir/Documents/MTTQ/Training/Task1_WeatherMan/code_files/report_calculator.py", line 11, in compute_extreme_stats
    highest_temp_reading = max(
TypeError: '>' not supported between instances of 'NoneType' and 'int'
tayyab.tahir@A006-01397 Task1_WeatherMan % python3
```